### PR TITLE
Feature/subscriptionbutton

### LINKED ISSUE #17: User Stories - Subscribe Button (8/25)
As a logged in user, I should be able to click on a crate in order to subscribe and be taken to the style preferences survey if I have not yet filled it out.

### PR DESCRIPTION
**Pull request by garnetred**
#### What's this PR do?
This PR fixes the issue with the axios post request so that we can receive a 200 response. It also adds functionality so that a user is redirected to the style survey from the subscriptions if they have not yet completed it, while also still correctly adding their new crate to the subscriptions page. The redux store is also updated to include the style preference. 

#### Where should the reviewer start?
Item.js, user/api/actions.js and user/api/state.js. 

#### How should this be manually tested?
This can be tested by loading the page in the browser, attempting to add a new crate, then completing the style survey, and finally viewing the updated subscriptions using the subscriptions item in the header. 

#### Any background context you want to provide?
We struggled a lot with the axios request partly due to formatting. There were also errors with the way we were trying to retrieve information from the response (using response.body.data instead of response.data.data). 

#### What are the relevant tickets?
closes #17

### CODE REVIEW

File: code/web/src/modules/user/api/actions.js [hunk 1]
```diff
@@ -1,120 +1,161 @@
...
+          stylePreference = response.data.data.addStyleToUser.style;
+          return dispatch({
+            type: SET_STYLE,
+            stylePreference,
+          });
+        }
       })
-  }
+      .catch((error) => console.log(error));
+  };
```
Comment (AntonioJacksonII) [inline code/web/src/modules/user/api/actions.js:102]: GREAT job getting this working after we all struggled for so long!

**Review by AntonioJacksonII — APPROVED** [review]

**Review by jpc20 — APPROVED** [review]
Awesome, way to go December!

File: code/web/src/modules/user/api/actions.js [hunk 2]
```diff
@@ -1,120 +1,161 @@
...
+          fields: ["id", "name", "email", "style"],
         })
+      )
+      .then((response) => {
+        if (response.data.data.addStyleToUser.style) {
+          stylePreference = response.data.data.addStyleToUser.style;
+          return dispatch({
+            type: SET_STYLE,
+            stylePreference,
+          });
```
Comment (Lithnotep) [inline code/web/src/modules/user/api/actions.js:98]: Awesome job getting axios working!

**Review by Lithnotep — APPROVED** [review]

File: code/web/src/modules/user/api/actions.js (hunk 1 above)
Comment (Melizzo) [reply code/web/src/modules/user/api/actions.js:102]: .then((response) => 
Did that allow the response to show up?
I also see that the response.data.data has an extra data - very cool - how did you figure that out?

File: code/web/src/modules/user/api/actions.js (hunk 2 above)
Comment (Melizzo) [reply code/web/src/modules/user/api/actions.js:98]: stylePreference - that makes sense of using just style

**Review by Melizzo — APPROVED** [review]

**Pull request merged**